### DISCUSSION TITLE
updating/updated hooks not called when array is emptied

### DISCUSSION THREAD

**Discussion by oli-laban** [livewire/livewire#8263] [1. Help] [answered]
I have a (currently) simple component with multiple checkboxes of categories. I need to dispatch an event when the categories change so I'm trying to use `updated` / `updatedSelectedCategories` to do that.

The hook is called as normal when selecting/changing the categories, however when deselecting the last category to end up with an empty array, the hook is not called. The same applies to the `updating` hook.

CategoryFilter.php:

```php
<?php

namespace App\Livewire\Events\Filters;

use App\Collections\EventCategoryCollection;
use App\Models\EventCategory;
use Livewire\Component;

class CategoryFilter extends Component
{
    public EventCategoryCollection $categories;

    /** @var int[] */
    public array $selectedCategories = [];

    public function mount(): void
    {
        $this->categories = EventCategory::select(['id', 'name'])->has('events')->get();
    }

    public function updated(): void
    {
        dd($this->selectedCategories);
    }
}
```

category-filter.blade.php:

```blade
<div>
    <div class="font-bold">Categories</div>
    @foreach($categories->take(4) as $category)
        <div>
            <label>
                <input type="checkbox" value="{{ $category->id }}" wire:model.live="selectedCategories">
                <span>{{ $category->name }}</span>
            </label>
        </div>
    @endforeach
    @if($categories->count() > 5)
        <div x-data="{ showAll: false }">
            <a href="#" x-show="!showAll" @click.prevent="showAll = true">See more</a>
            <div x-show="showAll">
                @foreach($categories->slice(4) as $category)
                    <div>
                        <label>
                            <input type="checkbox" value="{{ $category->id }}" wire:model.live="selectedCategories">
                            <span>{{ $category->name }}</span>
                        </label>
                    </div>
                @endforeach
                <a href="#" @click.prevent="showAll = false">See less</a>
            </div>
        </div>
    @endif
</div>
```

### Steps To Reproduce

Using above component:

1. Select some checkboxes
   - dd window will appear each time
2. Start deselecting checkboxes
   - dd window will appear each time
3. Deselect last checkbox
   - Nothing happens, though the array is correctly updated

**Comment by PerryvanderMeer** [comment, marked as answer]
Actually the `updated()` method will be called, try `dump()` instead of `dd()`.
When you return a non-succesful response (like `dd()`, which returns 500), you mess up with Livewire's javascript.

If you select a checkbox, then add `dd()` to your code (don't refresh!) and deselect the checkbox, you'll see also `dd()` is working.

**Reply by oli-laban** [reply]
Interesting, that does seem to be the case. Thanks!
